src/Freya.Integration/Integration.fs

```diff
@@ -22,36 +23,31 @@ module Freya.Integration
 open System
 open System.Threading.Tasks
 open Freya.Core
-
-(* OWIN Conversion *)
-
-/// Provides transformation functions for converting to/from OWIN from/to Freya.
-[<RequireQualifiedAccess>]
-[<CompilationRepresentation (CompilationRepresentationFlags.ModuleSuffix)>]
-module OwinAppFunc =
-    
-    /// Converts an <see cref="OwinAppFunc" /> to a <see cref="Freya{T}" /> computation
-    /// to allow use of standard OWIN components within Freya.
-    /// NOTE: EXPERIMENTAL
-    [<CompiledName ("ToFreya")>]
-    let toFreya (app: OwinAppFunc) : Freya<unit> =
-        fun s -> async {
-            let! token = Async.CancellationToken
-            // Apply and mutate the OwinEnvironment asynchronously
-            do! Async.AwaitTask <| app.Invoke(s.Environment).ContinueWith<unit>((fun _ -> ()), token)
-            // Return the result as a unit value and the mutated FreyaState
-            return (), s }
-
 open Freya.Pipeline
 
+(* OWIN Types *)
+
 /// Type alias for the OWIN MidFunc signature.
 type OwinMidFunc =
     Func<OwinAppFunc, OwinAppFunc>
 
+(* OWIN Conversion *)
+
 /// Provides transformation functions for converting to/from OWIN middlewares from/to Freya Pipelines.
 [<RequireQualifiedAccess>]
 [<CompilationRepresentation (CompilationRepresentationFlags.ModuleSuffix)>]
 module OwinMidFunc =
+
+    let private any (list: Async<'T>[])=
+        let tcs = new TaskCompletionSource<'T>()
+
+        list |> Array.map (fun wf->Async.Start (async{
+                    let! res=wf
+                    tcs.TrySetResult (res) |> ignore
+                }))
+             |> ignore
+
+        Async.AwaitTask tcs.Task
```

Comment:
Thank you, @haf! Shouldn't we move `any` to [`FSharpx.Async`](https://github.com/fsprojects/FSharpx.Async) and reference that as a dependency?